patches/tModLoader/Terraria/Player.TML.cs

```diff
@@ -594,4 +594,40 @@ public void AddImmuneTime(int cooldownCounterId, int immuneTime)
 			hurtCooldowns[cooldownCounterId] += immuneTime;
 		}
 	}
+
+	// Extra jumps
+	private ExtraJumpData[] extraJumps = new ExtraJumpData[ExtraJumpLoader.ExtraJumpCount];
+
+	public ref ExtraJumpData GetExtraJump<T>(T baseInstance) where T : ExtraJump => ref extraJumps[baseInstance.Type];
+
+	public ref ExtraJumpData GetExtraJump<T>() where T : ExtraJump => ref GetExtraJump(ModContent.GetInstance<T>());
+
+	public Span<ExtraJumpData> ExtraJumps => extraJumps.AsSpan();
+
+	internal Span<ExtraJumpData> ModdedExtraJumps => extraJumps[ExtraJumpLoader.DefaultExtraJumpCount..];
+
+	public bool AnyExtraJumpAvailable()
+	{
+		for (int i = 0; i < extraJumps.Length; i++)
+		{
+			ExtraJumpData data = extraJumps[i];
+			if (data.Active && data.JumpAvailable)
+				return true;
+		}
+
+		return false;
+	}
+
+	// An alternative more in line with the vanilla behavior
```

Comment:
Don't like the need for this, hard to reason about.

Reply:
this is just to keep the vanilla logic as it was in order to future proof against any inconsistencies (this is also why other quirks were left as-is)

Reply:
To keep this maintainable, we need to know why you would want to use this method rather than the one above. I understand "that's what vanilla had" but we're expanding on it here, so we need to be able to fully understand the system (vanilla isn't known for easy to understand code). 

I would expect that checking if jumps are available without an active check would result in potential bugs. Maybe the active check isn't _needed_ in the vanilla calls but it wouldn't hurt? I haven't looked at the logic deeply, but it seems spooky to me.

Reply:
in the vanilla logic, it's specifically checking if all extra jumps have been consumed (`JumpAvailable` would be `false` for all jumps).  furthermore, an extra jump is "consumed" by default

so the `Enabled` check would likely just end up with the same result?  after all, a disabled jump is always "consumed" and enabled jumps have to be used to be consumed... so checking `Enabled` would be pointless in a pure vanilla context

however, if modders can set `JumpAvailable` to true, then this would throw a wrench in the plan when the extra jump's `Enabled` flag is cleared

Reply:
resolved by removing the `NoActiveCheck` method in favor of making vanilla check `Enabled` as well via the `public` method